habitat/sims/habitat_simulator/habitat_simulator.py

```diff
@@ -257,41 +263,69 @@ def create_sim_config(
         )
 
         sensor_specifications = []
+        VisualSensorTypeSet = {
+            habitat_sim.SensorType.COLOR,
+            habitat_sim.SensorType.DEPTH,
+            habitat_sim.SensorType.SEMANTIC,
+        }
+        CameraSensorSubTypeSet = {
+            habitat_sim.SensorSubType.PINHOLE,
+            habitat_sim.SensorSubType.ORTHOGRAPHIC,
+        }
         for sensor in _sensor_suite.sensors.values():
-            sim_sensor_cfg = habitat_sim.SensorSpec()
-            # TODO Handle configs for custom VisualSensors that might need
-            # their own ignore_keys. Maybe with special key / checking
-            # SensorType
-            overwrite_config(
-                config_from=sensor.config,
-                config_to=sim_sensor_cfg,
-                # These keys are only used by Hab-Lab
-                # or translated into the sensor config manually
-                ignore_keys={
-                    "height",
-                    "hfov",
-                    "max_depth",
-                    "min_depth",
-                    "normalize_depth",
-                    "type",
-                    "width",
-                },
-            )
-            sim_sensor_cfg.uuid = sensor.uuid
-            sim_sensor_cfg.resolution = list(
-                sensor.observation_space.shape[:2]
-            )
-            sim_sensor_cfg.parameters["hfov"] = str(sensor.config.HFOV)
-
-            # TODO(maksymets): Add configure method to Sensor API to avoid
-            # accessing child attributes through parent interface
-            # We know that the Sensor has to be one of these Sensors
-            sensor = cast(HabitatSimVizSensors, sensor)
-            sim_sensor_cfg.sensor_type = sensor.sim_sensor_type
-            sim_sensor_cfg.gpu2gpu_transfer = (
-                self.habitat_config.HABITAT_SIM_V0.GPU_GPU
-            )
-            sensor_specifications.append(sim_sensor_cfg)
+
+            # Check if type VisualSensorSpec, we know that Sensor is one of HabitatSimRGBSensor, HabitatSimDepthSensor, HabitatSimSemanticSensor
+            if (
```

Comment:
Not all sensors are VisualSensor in Habitat.

Reply:
Not all sensors are VisualSensors but VisualSensorSpecs are the only kinds of specs that can be initialized at the moment.